core/domain/exp_services_test.py

```diff
@@ -7625,6 +7627,395 @@ def test_revert_exploration_with_mismatch_of_versions_raises_error(
             exp_services.revert_exploration('user_id', 'exp_id', 1, 0)
 
 
+class ExplorationTranslationCountTests(ExplorationServicesUnitTests):
+    """Test exploration translation count."""
+
+    def test_translation_count_upon_adding_new_interaction(self) -> None:
+        self.save_new_valid_exploration(
+            exploration_id=self.EXP_0_ID,
+            owner_id=self.owner_id,
+            end_state_name='End',
+            interaction_id='Continue',
+            content_html='A content')
+
+        exploration = exp_fetchers.get_exploration_by_id(self.EXP_0_ID)
+
+        opportunity_models.ExplorationOpportunitySummaryModel(
+            id=self.EXP_0_ID,
+            topic_id='topic_id',
+            topic_name='topic_name',
+            story_id='story_id',
+            story_title='story_title',
+            chapter_title='chapter_title',
+            content_count=3,
+            incomplete_translation_language_codes=[
+                language['id'] for language in (
+                    constants.SUPPORTED_AUDIO_LANGUAGES)],
+            language_codes_with_assigned_voice_artists=['en']
+        ).put()
+
+        translation_services.add_new_translation(
+            feconf.TranslatableEntityType.EXPLORATION, self.EXP_0_ID,
+            exploration.version, 'hi', 'content_0',
+            translation_domain.TranslatedContent(
+                'Translation 1',
+                translation_domain.TranslatableContentFormat.HTML,
+                False
+            )
+        )
+        translation_services.add_new_translation(
+            feconf.TranslatableEntityType.EXPLORATION, self.EXP_0_ID,
+            exploration.version, 'hi', 'ca_buttonText_2',
+            translation_domain.TranslatedContent(
+                'Translation 2',
+                translation_domain.TranslatableContentFormat.HTML,
+                False
+            )
+        )
+        translation_services.add_new_translation(
+            feconf.TranslatableEntityType.EXPLORATION, self.EXP_0_ID,
+            exploration.version, 'hi', 'content_3',
+            translation_domain.TranslatedContent(
+                'Translation 3',
+                translation_domain.TranslatableContentFormat.HTML,
+                False
+            )
+        )
```

Comment:
I think it would be good to verify transition counts after this step. Ditto elsewhere.

Reply:
Done